### PR TITLE
Home task7

### PR DESCRIPTION
**Pull request by AndrewLinkii**
fixed

### CODE REVIEW

**Comment by AndrewLinkii** [comment]
I'm sorry, somehow close the 
previous pullrequest

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 1]
```diff
@@ -6,14 +6,34 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int[] mainDiagonl = new int[matrix.length];
+        for (int i = 0; i < matrix.length && matrix[i].length > i; i++) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MatrixOperation.java]:
```suggestion
        for (int i = 0; i < matrix.length; i++) {
```
we use square matrix in this task so we shouldn't use this `&& matrix[i].length > i`

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 2]
```diff
@@ -6,14 +6,34 @@
...
+            mainDiagonl[i] = matrix[i][i];
+        }
+        return mainDiagonl;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        int tempMin = matrix[0][0];
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MatrixOperation.java]:
```suggestion
        int min = matrix[0][0];
```

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 3]
```diff
@@ -6,14 +6,34 @@
...
+                    tempMin = matrix[i][j];
+                }
+            }
+        }
+        return tempMin;
     }
 
     public int getMax(int[][] matrix) {
-        return 0;
+        int tempMax = matrix[0][0];
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MatrixOperation.java]:
```suggestion
        int max = matrix[0][0];
```

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 1 above)
Comment (AndrewLinkii) [reply src/main/java/core/basesyntax/MatrixOperation.java]: Rectangular matrix is ​​not necessarily square)

**Pull request closed without merging**